### PR TITLE
Integrate price feed with monitor bot

### PR DESCRIPTION
**Pull request by mrice32**

### CODE REVIEW

File: monitors/CRMonitor.js [hunk 1]
```diff
@@ -10,12 +10,16 @@ class CRMonitor {
   //    address: "0x12345",
   //    crAlert: 150 },
   // ...];
-  constructor(logger, expiringMultiPartyClient, walletsToMonitor) {
+  // @param priceFeed offchain price feed used to track the token price.
```
Comment (nicholaspai) [inline monitors/CRMonitor.js:13]: Can you document the other params, similar to the liquidator constructor comments?

File: monitors/CRMonitor.js [hunk 2]
```diff
@@ -27,14 +31,23 @@ class CRMonitor {
...
     // yield the price feed at the current time.
-    const contractTime = this.empClient.getLastUpdateTime();
-    const priceFeed = priceFunction(contractTime);
+    const price = this.priceFeed.getCurrentPrice();
+
+    if (!price) {
+      this.logger.warn({
+        at: "CRMonitor",
+        message: "Cannot compute wallet collateralization ratio because price feed returned invalid value",
+        price
```
Comment (chrismaree) [inline monitors/CRMonitor.js:42]: nit: is there a reason this is not key-valued like:
```
price: price
```

File: monitors/ContractMonitor.js [hunk 3]
```diff
@@ -97,14 +114,10 @@ class ContractMonitor {
     this.lastLiquidationBlockNumber = this.getLastSeenBlockNumber(latestLiquidationEvents);
   };
 
-  checkForNewDisputeEvents = async priceFunction => {
-    const contractTime = this.empEventClient.getLastUpdateTime();
-    const priceFeed = priceFunction(contractTime);
-
+  checkForNewDisputeEvents = async () => {
```
Comment (chrismaree) [inline monitors/ContractMonitor.js:117]: nice cleanup. These done need price

File: monitors/index.js [hunk 4]
```diff
@@ -28,22 +30,32 @@ const ExpandedERC20 = artifacts.require("ExpandedERC20");
...
-async function run(price, address, shouldPoll, botMonitorObject, walletMonitorObject, pollingDelay) {
+async function run(price, address, shouldPoll, botMonitorObject, walletMonitorObject, pollingDelay, priceFeedConfig) {
   Logger.info({
     at: "Monitor#index",
     message: "Monitor started 🕵️‍♂️",
     empAddress: address,
     currentPrice: price,
-    pollingDelay: pollingDelay
+    pollingDelay: pollingDelay,
+    priceFeedConfig
```
Comment (chrismaree) [inline monitors/index.js:40]: same nit as before: can this be key-valued?

File: monitors/index.js [hunk 5]
```diff
@@ -28,22 +30,32 @@ const ExpandedERC20 = artifacts.require("ExpandedERC20");
...
   });
 
   // Setup web3 accounts an contract instance
   const accounts = await web3.eth.getAccounts();
   const emp = await ExpiringMultiParty.at(address);
 
+  // Setup price feed.
+  // TODO: consider making getTime async and using contract time.
+  const getTime = () => Math.round(new Date().getTime() / 1000);
```
Comment (chrismaree) [inline monitors/index.js:49]: yup. you can get the contract time easily from the method that is already there using `getLastUpdateTime`

**Comment by chrismaree** [comment]
Also looks like your linter is failing.

File: monitors/CRMonitor.js (hunk 2 above)
Comment (mrice32) [reply monitors/CRMonitor.js:42]: You can write it like that if the current variable name is the same as the key you want to create. Happy to change if you have a stylistic preference to not use that sort of syntax.

**Comment by mrice32** [comment]
> Also looks like your linter is failing.

Weirdly my ternary is failing the linter and it won't autofix. I've also attempted to fix manually and can't get it to work. I may just remove the ternary to get it to work.

Thankfully, this is the first issue I've had with our new-ish linting system, so I'd say it's working very well overall.

File: monitors/CRMonitor.js (hunk 2 above)
Comment (chrismaree) [reply monitors/CRMonitor.js:42]: ok cool. We should do this in all places then.

File: monitors/CRMonitor.js (hunk 1 above)
Comment (mrice32) [reply monitors/CRMonitor.js]: Done.

**Pull request merged**